File: VB/ritem.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports DevExpress.XtraEditors.Repository
Imports DevExpress.XtraEditors.Registrator
Imports System.Drawing
Imports DevExpress.XtraEditors.Drawing

Namespace CustomRangeTrackBarControl
	<UserRepositoryItem("Register"), System.ComponentModel.DesignerCategory("None")> _
	Friend Class RepositoryItemMultipleRangeTrackBar
		Inherits RepositoryItemRangeTrackBar

		#Region "Register"
		Friend Const EditorName As String = "MultipleRangeTrackBar"

		Public Shared Sub Register()
            EditorRegistrationInfo.Default.Editors.Add(New EditorClassInfo(EditorName, GetType(MultipleRangeTrackBar), GetType(RepositoryItemMultipleRangeTrackBar), GetType(MultiRangeViewInfo), New MultiRangePainter(), True))
		End Sub
		Shared Sub New()
			Register()
		End Sub
		Public Overrides ReadOnly Property EditorTypeName() As String
			Get
				Return EditorName
			End Get
		End Property
		#End Region

		Public Overrides Function ToString() As String
			Return "Ritem_" & EditorName
		End Function
	End Class
End Namespace

File: CS/ritem.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using DevExpress.XtraEditors.Repository;
using DevExpress.XtraEditors.Registrator;
using System.Drawing;
using DevExpress.XtraEditors.Drawing;

namespace CustomRangeTrackBarControl
{
    [UserRepositoryItem("Register")]
    [System.ComponentModel.DesignerCategory("None")]
    class RepositoryItemMultipleRangeTrackBar : RepositoryItemRangeTrackBar
    {

        #region Register
        internal const string EditorName = "MultipleRangeTrackBar";

        public static void Register()
        {
            EditorRegistrationInfo.Default.Editors.Add(new EditorClassInfo(EditorName, typeof(MultipleRangeTrackBar),
                typeof(RepositoryItemMultipleRangeTrackBar), typeof(MultiRangeViewInfo),
                new MultiRangePainter(), true, null));
        }
        static RepositoryItemMultipleRangeTrackBar()
        {
            Register();
        }
        public override string EditorTypeName
        {
            get { return EditorName; }
        }
        #endregion

        public override string ToString()
        {
            return "Ritem_" + EditorName;
        }
    }
}
